chain-api/src/types/dtos.ts

```diff
@@ -283,15 +359,66 @@ export class ChainCallDTO {
     return copied;
   }
 
-  public isSignatureValid(publicKey: string): boolean {
```

Comment:
I think it would be far simpler to do: 

```ts
public isSignatureValid(publicKey: string, index: number = 0): boolean {
  const signatureObj: ChainCallDTO | SignatureDto | undefined = this.signatures.?[index] ?? (index == 0 ? this : undefined);
  if (signatureObj === undefined) {
    throw new ValidationError(`No signature at index ${index}`);
```

why:
- less changes in the existing code
- no `let`s and mutations (less tricky to analyze by human)

Reply:
(and also allows multiple signing schemes)